web-modal-sdk/immutableX/react-immutableX-web3auth-example/src/immutableRPC.ts

```diff
@@ -0,0 +1,122 @@
+import type { SafeEventEmitterProvider } from "@web3auth/base";
+import { ethers } from "ethers";
+
+import { ImmutableX, Config } from "@imtbl/core-sdk";
+import { createStarkSigner } from "@imtbl/core-sdk";
+
+export default class EthereumRpc {
+  private provider: SafeEventEmitterProvider;
+
+  constructor(provider: SafeEventEmitterProvider) {
+    this.provider = provider;
+  }
+
+  config = Config.SANDBOX; // Or Config.PRODUCTION or Config.ROPSTEN
+  client = new ImmutableX(this.config);
+
+  async getChainId(): Promise<any> {
+    try {
+      const ethersProvider = new ethers.providers.Web3Provider(this.provider);
+      // Get the connected Chain's ID
+      const networkDetails = await ethersProvider.getNetwork();
+      return networkDetails.chainId;
+    } catch (error) {
+      return error;
+    }
+  }
+
+  async getAccounts(): Promise<any> {
+    try {
+      const ethersProvider = new ethers.providers.Web3Provider(this.provider);
+      const ethSigner = ethersProvider.getSigner();
+
+      // Get user's Ethereum public address
+      const address = await ethSigner.getAddress();
+
+      // Get user's Starkex public address
+
+      const starkKey = await this.getPrivateKey();
+      // const starkKey = await generateStarkPrivateKey();
+      const starkSigner = createStarkSigner(starkKey);
+
+      return {
+        ethAddress: address,
+        starkAddress: starkSigner.getAddress(),
+      };
+    } catch (error) {
+      return error;
+    }
+  }
+
+  async registerAccounts(): Promise<any> {
+    try {
+      const ethersProvider = new ethers.providers.Web3Provider(this.provider);
+      const ethSigner = ethersProvider.getSigner();
+
+      // Get user's Ethereum public address
+      const address = await ethSigner.getAddress();
+
+      // Get user's Starkex public address
+
+      const starkKey = await this.getPrivateKey();
+      const starkSigner = createStarkSigner(starkKey);
+
+      console.log("eth address", address);
+      console.log("starkAddress", starkSigner);
+
+      const walletConnection = { ethSigner, starkSigner };
+      const response = await this.client.registerOffchain(walletConnection);
+      if (response.tx_hash) {
```

Comment:
What could be the `else` statement here? If this fails?

Should we address that?